### PR TITLE
feat: global refactoring

### PR DESCRIPTION
**Pull request by EnoRage**
- namings and 3rd party APIs removed (tenderly, swagger -> openapi, cli logic removed)
- abstract http executor interface with default impl at separate pkg (using common pkg to declare interface to split impl from usage)
- web3 provider interface that allows to execute web3 operations and can be switched to other option
- add Aggregation SDK as sep. package with new config and client structs 
- add Balances SDK as sep. package
- add Spot Prices SDK as sep. package
- add Gas Prices SDK as sep. package
- add Broadcast Transaction SDK as sep. package
- add NFT SDK as sep. package
- add History SDK as sep. package
- add Traces SDK as sep. package
- add Tokens SDK as sep. package

### CODE REVIEW

File: aggregation/swap_types_test.go [hunk 1]
```diff
@@ -1,13 +1,12 @@
-package models
+package aggregation
 
 import (
+	"github.com/1inch/1inch-sdk-go/internal/helpers/consts/chains"
+	"github.com/1inch/1inch-sdk-go/internal/helpers/consts/tokens"
```
Comment (Tanz0rz) [inline aggregation/swap_types_test.go:5]: Set your GoLand settings like this to do auto-grouping of the imports:

image

File: client/orderbook.go [hunk 2]
```diff
@@ -6,16 +6,16 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	helpers2 "github.com/1inch/1inch-sdk-go/internal/helpers"
```
Comment (Tanz0rz) [inline client/orderbook.go:9]: This import got aliased weirdly

File: internal/orderbook/limitorder.go [hunk 3]
```diff
@@ -3,6 +3,8 @@ package orderbook
 import (
 	"bufio"
 	"fmt"
+	helpers2 "github.com/1inch/1inch-sdk-go/internal/helpers"
```
Comment (Tanz0rz) [inline internal/orderbook/limitorder.go:6]: Another import alias issue

File: internal/validate/validate.go [hunk 4]
```diff
@@ -2,13 +2,12 @@ package validate
 
 import (
 	"fmt"
+	helpers2 "github.com/1inch/1inch-sdk-go/internal/helpers"
```
Comment (Tanz0rz) [inline internal/validate/validate.go:5]: again

**Review by Tanz0rz — COMMENTED** [review]
With the Tenderly simulations removed, we will need to be more precise with our unit testing. The first thing that comes to mind after our last conversation is to do local testing for permits (and fallback to approvals when the token does not support permits or something with the permit generation fails). 

I also want to note that doing real requests to the downstream APIs has provided value. Here is an example of a bug caught by sending requests to Limit Order API: https://1inch.atlassian.net/browse/DPA-914

File: aggregation/swap_types_test.go (hunk 1 above)
Comment (EnoRage) [reply aggregation/swap_types_test.go:5]: 👍

File: client/orderbook.go (hunk 2 above)
Comment (EnoRage) [reply client/orderbook.go]: helpers will be removed at all, as well as current file

File: internal/orderbook/limitorder.go (hunk 3 above)
Comment (EnoRage) [reply internal/orderbook/limitorder.go]: helpers will be removed at all, as well as current file

File: internal/validate/validate.go (hunk 4 above)
Comment (EnoRage) [reply internal/validate/validate.go]: helpers will be removed at all, as well as current file

**Review by Tanz0rz — APPROVED** [review]
I think we can merge this as it is now. There is still some important cleanup to, but this PR is getting too big and the current refactor is fully functional. 

Some notes for follow-up PRs:
- Improve codegen logic to make all query params concrete types even when they are optional
- Cleanup example program outputs
- Add additional APIs like Portfolio

**Pull request merged**